includes/wp-admin/class-advanced-settings-fields.php

```diff
@@ -90,6 +101,24 @@ public static function render_outbox_purge_days_field() {
 		) . '</p>';
 	}
 
+	/**
+	 * Render vary header field.
+	 */
+	public static function render_vary_header_field() {
+		$value = \get_option( 'activitypub_vary_header', '0' );
+		?>
+		<p>
+			<label>
+				<input type="checkbox" id="activitypub_vary_header" name="activitypub_vary_header" value="1" <?php checked( '1', $value ); ?> />
+				<?php echo \wp_kses( \__( 'Add the <code>Accept</code> Vary header for <code>application/activity+json</code> requests to the ActivityPub response.', 'activitypub' ), array( 'code' => array() ) ); ?>
+			</label>
+		</p>
+		<p class="description">
+			<?php \esc_html_e( 'This helps caching servers store the ActivityPub response based on different Accept headers. If you&#8217;re experiencing caching issues and seeing JSON responses in your browser, try enabling this option.', 'activitypub' ); ?>
```

Comment:
Wording suggestion that might be more user-friendly: "Enable this if you notice your site showing technical content instead of normal web pages, or if your ActivityPub connections seem unreliable. This setting helps your site deliver the right format of content to different services automatically."